### ISSUE TITLE
[From VSTS] FSharp emitting incorrect sequence point in debug info

### ISSUE THREAD

**Issue by cartermp** [Microsoft/visualfsharp#3036]
1. clone https://github.com/cartermp/15.2-E2E
2. Open in VS, F5
3. Set a breakpoint in PigLatin.fs, line 11
4. Type 'banana' in textbox, click generate button
5. You should now be stopped at a breakpoint
6. F5 to continue

**Expected:** continue past breakpoint

**Actual:** Stop at same breakpoint

**Additional Info:** It appears that the F# compiler is emitting an incorrect sequence point, causing the debugger to bind two separate breakpoints at the same source location.  The API `ISymUnmanagedMethod` is returning 30 sequence points for the ToPigLatin method, the first two are at IL offsets 0 and 7. Looking at the output of ildasm, it does not look like IL offset 7 should be a sequence point:

```
.method public static string  ToPigLatin(string str) cil managed
{
  // Code size       400 (0x190)
  .maxstack  6
  .locals init ([0] class [FSharp.Core]Microsoft.FSharp.Core.FSharpFunc`2<char,bool> isVowel,
           [1] char firstChar,
           [2] string V_2,
           [3] class [FSharp.Core]Microsoft.FSharp.Core.FSharpOption`1<int32> V_3,
           [4] class [FSharp.Core]Microsoft.FSharp.Core.FSharpOption`1<int32> V_4,
           [5] class [mscorlib]System.Tuple`2<int32,int32> V_5,
           [6] int32 V_6,
           [7] class [FSharp.Core]Microsoft.FSharp.Core.FSharpOption`1<int32> V_7,
           [8] int32 V_8,
           [9] class [FSharp.Core]Microsoft.FSharp.Core.FSharpOption`1<int32> V_9,
           [10] int32 V_10,
           [11] class [FSharp.Core]Microsoft.FSharp.Core.FSharpOption`1<int32> V_11,
           [12] class [FSharp.Core]Microsoft.FSharp.Core.FSharpOption`1<int32> V_12,
           [13] int32 V_13,
           [14] int32 V_14,
           [15] int32 V_15,
           [16] int32 V_16,
           [17] int32 V_17,
           [18] char V_18,
           [19] object V_19,
           [20] class [mscorlib]System.IFormattable V_20,
           [21] class [mscorlib]System.IFormattable V_21,
           [22] object V_22)
  .language '{AB4F38C9-B6E6-43BA-BE3B-58080B2CCCE3}', '{994B45C4-E6E9-11D2-903F-00C04FA302A1}', '{5A869D0B-6611-11D3-BD2A-0000F80849BD}'
// Source File 'C:\Users\chuckr\Source\Repos\15.2-E2E\E2E.Core\PigLatin.fs' 
//000011:         let firstChar = str.[0]
  IL_0000:  nop
  IL_0001:  newobj     instance void E2E.Core.PigLatin/isVowel@8::.ctor()
  IL_0006:  stloc.0
  IL_0007:  ldarg.0
  IL_0008:  ldc.i4.0
  IL_0009:  callvirt   instance char [mscorlib]System.String::get_Chars(int32)
  IL_000e:  stloc.1
//000012: 
//000013:         if isVowel firstChar then
  IL_000f:  ldloc.0
  IL_0010:  ldloc.1
  IL_0011:  callvirt   instance !1 class [FSharp.Core]Microsoft.FSharp.Core.FSharpFunc`2<char,bool>::Invoke(!0)
  IL_0016:  brfalse.s  IL_001a
  IL_0018:  br.s       IL_001c
  IL_001a:  br.s       IL_0028
//000014:             str + "way"
  IL_001c:  ldarg.0
  IL_001d:  ldstr      "way"
  IL_0022:  call       string [mscorlib]System.String::Concat(string,
                                                              string)
  IL_0027:  ret
```

**Comment by 0x53A** [comment]
I would like to take this opportunity and add an advertisement for a great tool: The latest dnSpy (https://ci.appveyor.com/project/0xd4d/dnspy/branch/master/artifacts) shows the sequence points inline in the IL view, which is awesome when you need to reason about that:

![image](https://cloud.githubusercontent.com/assets/4236651/26003136/225555f4-3732-11e7-9dec-bd84c6512741.png)

![image](https://cloud.githubusercontent.com/assets/4236651/26003310/9f8882f8-3732-11e7-8749-d3bdecbad6cb.png)

![image](https://cloud.githubusercontent.com/assets/4236651/26003301/9ae9c22a-3732-11e7-9d58-8b46c94e9ac1.png)

**Comment by dsyme** [comment]
Fixed by #3561
